### ISSUE TITLE
Dry Run Diff fails if previsouly created resource has been deleted

### ISSUE THREAD

**Issue by mitchellmaler** [pulumi/pulumi-kubernetes#1207]
### Problem description

I have enabled dryrun on our kubernetes provider which creates things like configmaps and secrets. If I delete the pulumi created resource such as a configmap it fails.

### Errors & Logs

 When it tries to build the patch I get the error 
```
Error: Failed to check for changes in resource RESOURCE_NAME because of an error computing the JSON patch describing the resource changes: configmaps "CONFIGMAP_NAME" not found
```

### Affected product version(s)

Pulumi kubernetes: 2.3.1 .NET
Pulumi 2.6.1

**Comment by lblackstone** [comment]
This is related to #955

**Comment by lblackstone** [comment]
This should be fixed by #988
